### DISCUSSION TITLE
Restore official LEGO® Firmware

### DISCUSSION THREAD

**Discussion by DamDinhHiep** [pybricks/support#1338] [Q&A] [answered]
I tried to restore LEGO's firmware.
I have completed the instructions on your website.
When I turned on the LEGO SPIKE software to install it, it froze.
![Untitled1](https://github.com/pybricks/support/assets/37657277/1835755f-f111-4e7e-9ed4-2d8ef61c2d8d)

**Comment by laurensvalk** [comment]
LEGO recently introduced some issues in the SPIKE V3 firmware update step. (It also happens with their regular updates, not just after recovery. They have confirmed this on their technical support page.)

You may wish to install the official V2 firmware first via https://spikelegacy.legoeducation.com/ and then upgrade to V3.

I'll try this myself later and see if we can provide an update to fix theirs.

**Comment by laurensvalk** [comment, marked as answer]
Thanks for your note @DamDinhHiep.

The following recipe works:

1. Restore the firmware via https://code.pybricks.com/
2. Follow the steps on [this page](https://dfu.pybricks.com/fix-update/) to fix issues with the LEGO installer. (You’re done when the light turns pink; no need to use the Mindstorms app; you’ll use the Spike app in the next step) 
3. Use the official [LEGO SPIKE app](https://spike.legoeducation.com/) to update the hub.

I have just verified these steps to make sure that they work.

Step 2 is normally not required, but it helps work around an issue in the official LEGO app. We'll let LEGO know about it. Thanks for reporting!

**Reply by DamDinhHiep** [reply]
I tried it.
I have finished restoring the LEGO SPIKE firmware.
Thank you very much.
